### DISCUSSION TITLE
Declarative Jenkins Pipeline examples for sending DT project vulnerability reports

### DISCUSSION THREAD

**Discussion by aejazuddinam** [DependencyTrack/dependency-track#1163] [Q&A]
Dear All,

Can anyone please help me with examples of declarative pipeline example of sending emails whenever a new bom is uploaded in the project and can we send the project vulnerability report in email ?

I have checked with the DT Dashboard UI email option, but i want to configure it from declarative pipeline for complete automation not from UI.

Regards,
Aejaz

**Comment by stevespringett** [comment]
Use the swagger docs to see what the API looks like, or inspect the request/response in your browser using the built-in development tools.

**Reply by aejazuddinam** [reply]
Creating, Updating alerts, updating publisher, notification level etc are present with the APIs. 

I dont see if there are any functionality like after bom is imported in any project and vulnerability are shown in UI, can we get a report of that complete project vulnerability to email address rather than looking in UI.

Any functionality also to get CSV or HTML etc reports for vulnerabilities of DT in email after bom is imported

**Reply by stevespringett** [reply]
Reports (CSV, HTML, PDF, etc) are not supported nor planned. Many users use Kenna Security, Fortify SSC, ThreatFix, DefectDojo, etc to pull in data, aggregate it, and run reports on it. Other users use general purpose business intelligence tools (DOMO, Tableau, Apache SuperSet, etc) to pull in all their security data and run reports against it.
